sdk/scheduler/src/main/java/com/mesosphere/sdk/state/FrameworkStore.java

```diff
@@ -0,0 +1,103 @@
+package com.mesosphere.sdk.state;
+
+import java.util.Optional;
+
+import org.apache.mesos.Protos;
+import org.slf4j.Logger;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.mesosphere.sdk.offer.LoggingUtils;
+import com.mesosphere.sdk.storage.Persister;
+import com.mesosphere.sdk.storage.PersisterException;
+import com.mesosphere.sdk.storage.StorageError.Reason;
+
+/**
+ * A {@code FrameworkStore} stores the Framework ID for the scheduler. The framework ID is required for the scheduler to
+ * re-identify itself with Mesos after initial registration.
+ *
+ * <p>The structure used in the underlying persister is as follows:
+ * <br>rootPath/
+ * <br>&nbsp;-> FrameworkID
+ */
+public class FrameworkStore {
+
+    private static final Logger LOGGER = LoggingUtils.getLogger(FrameworkStore.class);
+    private static final String FWK_ID_PATH_NAME = "FrameworkID";
+
+    private final Persister persister;
+
+    /**
+     * Creates a new {@link FrameworkStore} which uses the provided {@link Persister} to access the Framework ID.
+     *
+     * @param persister The persister which holds the data
+     */
+    public FrameworkStore(Persister persister) {
+        this.persister = persister;
+    }
+
+    /**
+     * Returns the underlying {@link Persister}.
+     */
+    public Persister getPersister() {
```

Comment:
When would something need to access the persister?

Reply:
I think this (will) come up when we're clearing the persister data during uninstall, or something like that.